pkg/detectors/gemini/gemini.go

```diff
@@ -0,0 +1,138 @@
+package gemini
+
+import (
+	"bytes"
+	"context"
+	"crypto/hmac"
+	"crypto/sha512"
+	"encoding/base64"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/trufflesecurity/trufflehog/v3/pkg/common"
+	"github.com/trufflesecurity/trufflehog/v3/pkg/detectors"
+	"github.com/trufflesecurity/trufflehog/v3/pkg/pb/detectorspb"
+)
+
+type Scanner struct{}
+
+const (
+	baseURL       = "https://api.gemini.com"
+	accountDetail = "/v1/account"
+	account       = "primary"
+)
+
+// Ensure the Scanner satisfies the interface at compile time.
+var _ detectors.Detector = (*Scanner)(nil)
+
+var (
+	client = common.SaneHttpClient()
+
+	// Make sure that your group is surrounded in boundary characters such as below to reduce false positives.
+	keyPat    = regexp.MustCompile(`\b((?:master-|account-)[0-9A-Za-z]{20})\b`)
+	secretPat = regexp.MustCompile(`[A-Za-z0-9]{27,28}`)
+)
+
+// Keywords are used for efficiently pre-filtering chunks.
+// Use identifiers in the secret preferably, or the provider name.
+func (s Scanner) Keywords() []string {
+	return []string{"master-", "account-"}
+}
+
+// FromData will find and optionally verify Gemini secrets in a given set of bytes.
+func (s Scanner) FromData(_ context.Context, verify bool, data []byte) (results []detectors.Result, err error) {
+	dataStr := string(data)
+
+	idMatches := keyPat.FindAllStringSubmatch(dataStr, -1)
+	secretMatches := secretPat.FindAllStringSubmatch(dataStr, -1)
+
+	for _, match := range idMatches {
+		if len(match) != 2 {
+			continue
+		}
+		resMatch := strings.TrimSpace(match[1])
+
+		for _, secretMatch := range secretMatches {
+			resSecretMatch := strings.TrimSpace(secretMatch[0])
+
+			s1 := detectors.Result{
+				DetectorType: detectorspb.DetectorType_Gemini,
+				Raw:          []byte(resMatch),
+				RawV2:        []byte(resMatch + resSecretMatch),
+			}
+
+			if verify {
+				req, err := constructRequest(resSecretMatch, resMatch)
+				if err != nil {
+					continue
+				}
+
+				func() {
+					res, err := client.Do(req)
+					if err == nil {
+						defer res.Body.Close()
+						if res.StatusCode >= 200 && res.StatusCode < 300 {
+							s1.Verified = true
+							return
+						}
+						// This function will check false positives for common test words, but also it will make sure the key appears 'random' enough to be a real key.
+						if detectors.IsKnownFalsePositive(resSecretMatch, detectors.DefaultFalsePositives, true) {
+							return
+						}
```

Comment:
Yep the default should be false. Are you suggesting set it explicitly?

Reply:
Er, this line effectively does nothing since it's already at the end of the function. The line before is setting `s1.Verified` to `true` so if we find it's a false positive then `s1.Verified` is still `true`.

Should we maybe do the false positive check earlier to save a network call?

Reply:
Oh but if it's verified we don't do the false positive check since we return. We only do false positive check if status is <200 or >=300.

Reply:
Or maybe i'm missing something here 🤔

Reply:
oooof... yep you are correct. There was something weird about this detector. I was so close to using a goto here 🤣